libretroshare/src/pqi/pqistreamer.cc

```diff
@@ -199,18 +201,30 @@ RsItem *pqistreamer::GetItem()
 	pqioutput(PQL_DEBUG_ALL, pqistreamerzone, "pqistreamer::GetItem()");
 #endif
 
-	RsStackMutex stack(mStreamerMtx); /**** LOCKED MUTEX ****/
```

Comment:
Removing that lock seems very risky to me, and also not needed anyway: the time spent in taking one element from the queue is negligible. So I suppose you want to avoid waiting for the lock.

Reply:
We don't save much by removing this lock, but it is still 100pct useless.
There is no competition at all for the mIncoming queue: there is only 1 thread accessing it (pushing in or poping from).